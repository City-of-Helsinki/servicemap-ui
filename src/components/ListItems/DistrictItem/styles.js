export default theme => ({
  simpleItem: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'start',
    paddingLeft: theme.spacing(3),
    paddingTop: theme.spacing(1.5),
    paddingBottom: theme.spacing(1.5),
  },
  simpleTitle: {
    marginBottom: theme.spacing(0.5),
  },
  itemTextContainer: {
    display: 'flex',
    alignItems: 'center',
  },
  boldText: {
    fontWeight: 'bold',
  },
  areaIcon: {
    fontSize: '1.25rem',
    marginLeft: 0,
    marginRight: theme.spacing(2),
  },
  divider: {
    listStyleType: 'none',
  },
  padding: {
    paddingLeft: theme.spacing(3),
  },
});
