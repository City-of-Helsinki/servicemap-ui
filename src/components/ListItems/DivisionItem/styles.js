export default theme => ({
  containerInner: {
    display: 'flex',
    flexDirection: 'row',
    padding: theme.spacing(0, 1),
  },
  content: {
    alignSelf: 'center',
    flex: '1 1 auto',
  },
  divisionAddress: {
    color: 'black',
    display: 'block',
    fontWeight: 'normal',
  },
  divisionDistance: {
    color: 'black',
    fontWeight: 'normal',
  },
  divisionTitle: {
    fontWeight: 'bold',
    paddingLeft: theme.spacing(2),
    marginBottom: theme.spacing(1),
  },
  emergencyContent: {
    display: 'flex',
    flexDirection: 'row',
    padding: theme.spacing(0, 1),
    marginLeft: theme.spacing(5),
    marginRight: 60,
  },
  emergencyTypography: {
    color: 'black',
    fontWeight: 'normal',
  },
  flexRow: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  icon: {
    marginRight: theme.spacing(2),
  },
  listItem: {
    alignItems: 'stretch',
    display: 'flex',
    flexDirection: 'column',
    padding: theme.spacing(1),
  },
  li: {
    listStyleType: 'none',
  },
  linkButton: {
    flexDirection: 'column',
    alignItems: 'initial',
    '&:hover': {
      textDecoration: 'underline',
    },
  },
  weightBold: {
    fontWeight: 'bold',
  },
});
