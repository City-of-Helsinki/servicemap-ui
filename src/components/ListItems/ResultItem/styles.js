
export default theme => ({
  listItemPadded: {
    padding: theme.spacing.unitDouble,
  },
  listItem: {
    padding: theme.spacing.unit,
  },
  listItemIcon: {
    margin: theme.spacing.unit,
    marginRight: theme.spacing.unitDouble,
  },
  listItemIconPadded: {
    margin: theme.spacing.unit,
  },
  bottomContainer: {
    display: 'inline-block',
    marginTop: '6px',
    '& p': {
      fontWeight: 'normal',
    },
  },
  bottomHighlight: {
    backgroundColor: '#323232',
    color: '#FFF',
    padding: '0 8px',
    marginBottom: '2px',
  },
  cssFocused: {
    outlineStyle: 'solid',
    outlineColor: 'blue',
    outlineWidth: 2,
  },
  title: {
    flex: '1 1 auto',
    textOverflow: 'ellipsis',
    margin: 0,
  },
  noMargin: {
    margin: 0,
  },
  itemTextContainer: {
    flex: '1 1 auto',
    margin: 0,
    marginLeft: theme.spacing.unitDouble,
  },
  topRow: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
  },
  bottomRow: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
  },
  marginLeft: {
    marginLeft: theme.spacing.unit,
  },
  rightColumn: {
    textAlign: 'right',
  },
  caption: {
    color: 'rgba(0,0,0,0.6)',
  },
  text: {
    color: '#000',
    fontWeight: 'normal',
  },
  divider: {
    marginRight: -theme.spacing.unitDouble,
  },
});
