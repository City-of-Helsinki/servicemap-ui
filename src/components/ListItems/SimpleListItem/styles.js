export default theme => ({
  listItem: {
    minHeight: '3.5rem',
    padding: theme.spacing.unit,
  },
  textContainer: {
    padding: `${theme.spacing.unit}px 0`,
    marginLeft: theme.spacing.unitDouble,
    marginRight: theme.spacing.unitDouble,
    whiteSpace: 'pre-line',
  },
  link: {
    color: '#0000EE',
  },
  listIcon: {
    width: '1.5rem',
    height: '1.5rem',
    margin: theme.spacing.unit,
    marginRight: theme.spacing.unitDouble,
  },
  divider: {
    marginLeft: theme.spacing.unit * 9,
    marginRight: -theme.spacing.unitDouble,
  },
});
