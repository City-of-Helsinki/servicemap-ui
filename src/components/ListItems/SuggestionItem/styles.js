export default theme => ({
  container: {
    display: 'flex',
    flex: '1 1 auto',
    padding: `${theme.spacing.unitHalf}px 0`,
  },
  listItem: {
    padding: `0 ${theme.spacing.unit}px`,
  },
  textContainer: {
    display: 'flex',
    padding: `${theme.spacing.unit}px 0`,
    marginLeft: theme.spacing.unitDouble,
    marginRight: theme.spacing.unitDouble,
    whiteSpace: 'pre-line',
    '& p': {
      lineHeight: '18px',
    },
  },
  listIcon: {
    width: '1.5rem',
    height: '1.5rem',
    margin: `${theme.spacing.unitHalf}px ${theme.spacing.unit}px`,
    marginRight: 0,
    alignSelf: 'center',
    padding: theme.spacing.unitHalf,
  },
  suggestIcon: {
    color: '#757575',
    height: 'auto',
    margin: 0,
    padding: 0,
  },
  suggestIconLabel: {
    padding: theme.spacing.unit,
  },
  subtitle: {
    fontSize: '10px',
    fontWeight: 'none',
    lineHeight: '18px',
  },
  divider: {
    marginLeft: theme.spacing.unit * 8,
    marginRight: -theme.spacing.unitDouble,
  },
  text: {
    alignSelf: 'center',
  },
});
