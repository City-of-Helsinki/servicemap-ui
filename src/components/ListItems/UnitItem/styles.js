export default () => ({
  title: {
    fontWeight: 'bold',
  },
});
