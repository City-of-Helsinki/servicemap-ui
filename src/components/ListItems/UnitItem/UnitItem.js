/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import UnitHelper from '../../../utils/unitHelper';
import ResultItem from '../ResultItem';
import SettingsUtility from '../../../utils/settings';
import UnitIcon from '../../SMIcon/UnitIcon';
import isClient, { uppercaseFirst } from '../../../utils';

const UnitItem = ({
  classes,
  distance,
  unit,
  onClick,
  getLocaleText,
  intl,
  padded,
  navigator,
  settings,
}) => {
  const parseAccessibilityText = () => {
    const accessSettingsSet = SettingsUtility.hasActiveAccessibilitySettings(settings);
    let accessText = null;
    let accessibilityProblems = null;
    if (accessSettingsSet && unit && settings) {
      accessibilityProblems = UnitHelper.getShortcomingCount(unit, settings);
      accessText = intl.formatMessage({ id: 'unit.accessibility.noInfo' });
      if (accessibilityProblems !== null && typeof accessibilityProblems !== 'undefined') {
        switch (accessibilityProblems) {
          case 0:
            accessText = intl.formatMessage({ id: 'unit.accessibility.ok' });
            break;
          default:
            accessText = intl.formatMessage({ id: 'unit.accessibility.problems' }, { count: accessibilityProblems });
        }
      }
    }
    return { text: accessText, problemCount: accessibilityProblems };
  };

  // Don't render if not valid unit
  if (!UnitHelper.isValidUnit(unit)) {
    return null;
  }

  const icon = isClient() ? <UnitIcon unit={unit} /> : null;

  // Parse unit data
  const {
    contract_type, id, name,
  } = unit;

  // Accessibility text and color
  const accessData = isClient() ? parseAccessibilityText() : 0;
  const accessText = accessData.text;
  const { problemCount } = accessData;

  // Contract type text
  const contractType = contract_type && contract_type.description ? uppercaseFirst(getLocaleText(contract_type.description)) : '';
  const distanceText = distance ? {
    text: `${distance.distance} ${distance.type}`,
    srText: `${distance.distance} ${distance.type === 'm'
      ? intl.formatMessage({ id: 'general.distance.meters' })
      : intl.formatMessage({ id: 'general.distance.kilometers' })}`,
  } : {};

  return (
    <ResultItem
      title={getLocaleText(name)}
      subtitle={contractType}
      bottomText={accessText}
      bottomHighlight={problemCount !== null && typeof problemCount !== 'undefined'}
      extendedClasses={{
        typography: {
          title: classes.title,
        },
      }}
      distance={distanceText}
      icon={icon}
      onClick={(e) => {
        e.preventDefault();
        if (onClick) {
          onClick();
        } else if (navigator) {
          navigator.push('unit', { id });
        }
      }}
      padded={padded}
    />
  );
};

export default UnitItem;

// Typechecking
UnitItem.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  distance: PropTypes.shape({
    distance: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    type: PropTypes.oneOf(['m', 'km']),
    text: PropTypes.string,
  }),
  unit: PropTypes.objectOf(PropTypes.any),
  getLocaleText: PropTypes.func.isRequired,
  onClick: PropTypes.func,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  navigator: PropTypes.objectOf(PropTypes.any),
  settings: PropTypes.objectOf(PropTypes.any).isRequired,
  padded: PropTypes.bool,
};

UnitItem.defaultProps = {
  distance: null,
  unit: {},
  onClick: null,
  navigator: null,
  padded: false,
};
