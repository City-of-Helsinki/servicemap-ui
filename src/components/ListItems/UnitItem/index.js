import UnitItem from './UnitItem';

export default UnitItem;
