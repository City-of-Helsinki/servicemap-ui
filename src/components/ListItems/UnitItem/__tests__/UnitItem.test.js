import React from 'react';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import themes from '../../../../themes';
import { initialState } from '../../../../redux/reducers/user';
import UnitItem from '../index';

const mockData = {
  accessibility_properties: [],
  accessibility_shortcoming_count: {},
  contract_type: {
    id: 'municipal_service',
    description: { fi: 'kunnallinen palvelu', sv: 'kommunal tjänst', en: 'municipal service' },
  },
  id: 63115,
  municipality: 'espoo',
  name: { fi: 'Lippulaivan kirjasto', sv: 'Lippulaivabiblioteket', en: 'Lippulaiva library' },
  object_type: 'unit',
  street_address: { fi: 'Merikarhunkuja 11', sv: 'Sjöbjörnsgränden 11', en: 'Merikarhunkuja 11' },
};

// Generic required props for SimpleListItem
const mockProps = {
  unit: mockData,
  onClick: () => {},
  simpleItem: false,
};

// Mock props for intl provider
const intlMock = {
  locale: 'en',
  messages: {
    'unit.accessibility.noInfo': 'No info',
    'unit.accessibility.ok': 'Accessibility ok',
    'unit.accessibility.problems': '{count} accessibility problems'
  },
};

// simpleItem={embeddedList} key={`unit-${id}`} className={`unit-${id}`} unit={item}

const mockStore = configureStore([]);

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => {
  const store = mockStore({
    user: initialState,
    settings: {},
  });

  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
};

const renderWithProviders = component => render(component, { wrapper: Providers });

describe('<UnitItem />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<UnitItem {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does render accessibility attributes correctly', () => {
    const { container } = renderWithProviders(<UnitItem {...mockProps} />);
    const items = container.querySelectorAll('li');
    const firstItem = items[0];
    const firstItemSRText = firstItem.querySelectorAll('p')[0];
    const firstItemResultTitle = firstItem.querySelectorAll('p')[1];
    const dividerItem = items[1];

    // List item's image should be aria-hidden
    expect(firstItem.querySelector('img').getAttribute('aria-hidden')).toBeTruthy();
    expect(firstItem.getAttribute('role')).toEqual('link');
    expect(firstItem.getAttribute('tabIndex')).toEqual('0');
    expect(firstItemSRText.className.indexOf('ResultItem-title') > 0).toBeTruthy();
    expect(firstItemResultTitle.className.indexOf('ResultItem-title') > 0).toBeTruthy();
    expect(firstItemResultTitle.getAttribute('aria-hidden')).toBeTruthy();
    expect(dividerItem.getAttribute('aria-hidden')).toBeTruthy();
  });
});
