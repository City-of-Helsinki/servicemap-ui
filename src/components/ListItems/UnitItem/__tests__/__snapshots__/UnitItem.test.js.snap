// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnitItem /> should work 1`] = `
<div>
  <li
    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button ResultItem-listItem-3 css-16e6fe9-MuiButtonBase-root-MuiListItem-root"
    role="link"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root ResultItem-listItemIcon-4 css-cveggr-MuiListItemIcon-root"
    >
      <img
        alt=""
        aria-hidden="true"
        class="Connect(UnitIcon)-icon-21 "
        src="data:image/png;base64,00"
      />
    </div>
    <div
      class="ResultItem-itemTextContainer-10  "
    >
      <div
        class="ResultItem-topRow-12"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 ResultItem-title-8 ResultItem-srOnly css-1ak20dk-MuiTypography-root"
          style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
        >
          
    Lippulaivan kirjasto 
    Kunnallinen palvelu 
     
     
    
  
        </p>
        <p
          aria-hidden="true"
          class="MuiTypography-root MuiTypography-body2 ResultItem-title-8  Connect(UnitItem)-title-1  ResultItem-title css-1ak20dk-MuiTypography-root"
          role="textbox"
        >
          Lippulaivan kirjasto
        </p>
      </div>
      <div>
        <div>
          <p
            aria-hidden="true"
            class="MuiTypography-root MuiTypography-caption ResultItem-noMargin-9 ResultItem-text-17  ResultItem-subtitle css-r3hlsg-MuiTypography-root"
          >
            Kunnallinen palvelu
          </p>
        </div>
      </div>
    </div>
  </li>
  <li
    aria-hidden="true"
  >
    <hr
      class="MuiDivider-root MuiDivider-inset ResultItem-divider-18 css-uls1yk-MuiDivider-root"
    />
  </li>
</div>
`;
